ModularTegustation/tegu_items/library/library.dm

```diff
@@ -0,0 +1,534 @@
+	var/list/waw_list = list(
+		/mob/living/simple_animal/hostile/abnormality/apex_predator,
+		/mob/living/simple_animal/hostile/abnormality/black_swan,
+		/mob/living/simple_animal/hostile/abnormality/clouded_monk,
+		/mob/living/simple_animal/hostile/abnormality/clown,
+		/mob/living/simple_animal/hostile/abnormality/ebony_queen,
+		/mob/living/simple_animal/hostile/abnormality/hatred_queen,
+		/mob/living/simple_animal/hostile/abnormality/nosferatu,
+		/mob/living/simple_animal/hostile/abnormality/thunder_bird,
+		/mob/living/simple_animal/hostile/abnormality/warden
+	)
+
+	// Dreaming is here because it's too easy, hey free waw
+	var/list/he_list = list(
+		/mob/living/simple_animal/hostile/abnormality/dreaming_current,
+		/mob/living/simple_animal/hostile/abnormality/blue_shepherd,
+		/mob/living/simple_animal/hostile/abnormality/funeral,
+		/mob/living/simple_animal/hostile/abnormality/golden_apple,
+		/mob/living/simple_animal/hostile/abnormality/headless_ichthys,
+		/mob/living/simple_animal/hostile/abnormality/helper,
+		/mob/living/simple_animal/hostile/abnormality/kqe,
+		/mob/living/simple_animal/hostile/abnormality/pinocchio,
+		/mob/living/simple_animal/hostile/abnormality/pisc_mermaid,
+		/mob/living/simple_animal/hostile/abnormality/puss_in_boots,
+		/mob/living/simple_animal/hostile/abnormality/red_buddy,
+		/mob/living/simple_animal/hostile/abnormality/scarecrow,
+		/mob/living/simple_animal/hostile/abnormality/wayward,
+		/mob/living/simple_animal/hostile/abnormality/woodsman
+	)
+
+	var/list/teth_list = list(
+		/mob/living/simple_animal/hostile/abnormality/fairy_longlegs,
+		/mob/living/simple_animal/hostile/abnormality/forsaken_murderer,
+		/mob/living/simple_animal/hostile/abnormality/fragment,
+		/mob/living/simple_animal/hostile/abnormality/redblooded,
+		/mob/living/simple_animal/hostile/abnormality/smile
+	)
+
+	var/list/risk_list = list(
+	)
+
+/obj/library_director/Initialize()
+	. = ..()
+	GLOB.Pale_Librarian = src
+	floor_list = list(
+		general_floor, history_floor, technology_floor, literature_floor, art_floor, 
+		natural_floor, language_floor, social_floor, philosophy_floor, religion_floor
+	)
+
+	risk_list = list(
+		aleph_list, waw_list, he_list, teth_list
+	)
+
+/obj/library_director/proc/GenerateReception()
+	if(!LAZYLEN(selected_abnos))
+		PickAbno()
+	if(!selected_floor)
+		PickFloor()
+	var/TP_Landmark
+	for(var/obj/effect/landmark/library_floor/LF in GLOB.landmarks_list)
+		TP_Landmark = get_turf(LF)
+	var/datum/map_template/shelter/template = SSmapping.shelter_templates[selected_floor]
+	template.load(TP_Landmark, centered = TRUE)
+	Explain()
+
+/obj/library_director/proc/Explain()
+	if(LAZYLEN(special_greetings))
+		for(var/i in special_greetings)
+			sleep(15)
+			say(i)
+	else
+		for(var/i in greetings)
+			sleep(15)
+			say(i)
+	
+//Calculates the user level and armor values, weapons don't have much consistency to them (BL is quite fucked either way)
+/obj/library_director/proc/PickAbno()
```

Comment:
I kinda assume that it doesn't get more than like 5 receptions per shift so that it never goes empty
Also there is no reason to strip your armor since you just get worse rewards if you do